flybywiresim/aircraft
# feat(a380x): proper trim tank calculation/logic for automatic refueling

Fixes #8827

## Summary of Changes

See issue

* Updated FLT spawn states for new fuel logic

## Screenshots (if necessary)



![image](https://github.com/user-attachments/assets/0034ebf5-066c-4c0a-9e42-7576ec722dea)


## References





## Additional context



Discord username (if different from GitHub):

## Testing instructions

Test when spawning in with a brand new ATC ID (aka use a livery you have not already used before. If you do not have one, get a new livery)

- Fuel state should spawn in about half full, with trim tank half full (9000-11000KG)
   -  If you do not have a new livery to test with, then it should remember the saved fuel state of your previous flight, so this instruction would not apply. 
- Without loading any payload, instantly load fuel to a lower or higher value (but not entirely empty/full).
   - Trim tank should update, and GW CG MAC % (See on payload tab, switch from ZFW CGMAC%) should display around 38-42 CGMAC %. 
   - The logic attempts to target 39.5% CG MAC, but there is some variance.
   - Note: When closer to empty, GW CG MAC % will tend to shift forward of this mark (>38%). That is normal and expected.
- Load some payload (i.e. 400 pax/cargo), then go back and adjust the fuel level again, then load. 
   - GW CGMAC % should remain around 38-42%
- Create an OFP flight plan, then import it. into the EFB. Go to the payload and fuel tabs, and load them in any order. 
   - GW CGMAC % should remain around 38-42%
   
- Regression test: using Fast and Real Time fueling still works as expected.


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page
